src/types/interfaces.ts

```diff
@@ -230,101 +230,85 @@ export interface Layout extends NameValuePair {
   containerName?: string;
 }
 
+interface StateConfig<T> {
```

Comment:
Where do you think we can capture out TS patterns and best practices like noting that we use T as a placeholder for another type that will be specified when StateConfig is used.  or are some things just a generic convention?